cirq-google/cirq_google/api/v2/run_context.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence
+from cirq_google.api.v2 import program_pb2
+from cirq_google.api.v2 import run_context_pb2
+
+
+# The special index of an empty directory path [].
+_EMPTY_DIR_PATH_IDX = -1
+
+
+def to_device_parameters_diff(
```

Comment:
My understanding is that users never interact with `run_context` directly. Instead they call [`run_sweep`](https://github.com/quantumlib/Cirq/blob/3080d939ef5cf2a70168e8fbdf085c8f0b214387/cirq-google/cirq_google/engine/engine_processor.py#L125) and pass in `cirq.Sweepable` and the number of `repetitions`, and we [construct](https://github.com/quantumlib/Cirq/blob/3080d939ef5cf2a70168e8fbdf085c8f0b214387/cirq-google/cirq_google/engine/engine.py#L350 ) the run_context on their behalf.

It would be nice to have a similar `cirq.DeviceParametesDiff` concept that the user can pass into `run_sweep` that can later be [serialized](https://github.com/quantumlib/Cirq/blob/3080d939ef5cf2a70168e8fbdf085c8f0b214387/cirq-google/cirq_google/engine/engine.py#L141) into the run_context. WDYT?

Reply:
the proto is for a user to construct a RunJobRequest, in order to invoke RunJob on a cirq server.
the DeviceParametersDiff proto is not intended for uses to make python local call to run_sweep.

let me clarify it in the pr description.

Reply:
oh, I was incorrectly assuming you were referring to our internal run_sweep API, https://github.com/qh-lab/pyle/blob/master/src/pyle/dataking/cirq_sweep/_base.py#L87 , and was not aware there is a similarly named function in cirq-google repo.

yes, I will need to make changes to the cirq run_sweep function to incorporate an optional device parameters diff concept. would it be viable to do it later, since we will need to make changes to cirq server first, before exposing the device parameeters overriding in run_sweep call?